nextjs_tdd/__tests__/unit/_app/@home/home.test.tsx

```diff
@@ -0,0 +1,32 @@
+import Home from '@/app/@home/home';
+import { COMMON_CONST } from '@/constants/app-constants';
+import '@testing-library/jest-dom';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { mockCity } from '../../../__fixtures__/home';
+import ContextWrapper from '@/context';
+
+describe('Page', () => {
+    test('should render tittle properly', () => {
+        render(
+            <ContextWrapper>
+                <Home />
+            </ContextWrapper>
+        );
+        const cityPlaceholder = screen.getByPlaceholderText(COMMON_CONST.CITY_PLACEHOLDER);
+        expect(cityPlaceholder).toBeInTheDocument();
+    });
+
+    test('should render tittle properly', () => {
+        render(
+            <ContextWrapper>
+                <Home />
+            </ContextWrapper>
+        );
+        const cityPlaceholder = screen.getByPlaceholderText(COMMON_CONST.CITY_PLACEHOLDER);
+        fireEvent.change(screen.getByPlaceholderText(COMMON_CONST.CITY_PLACEHOLDER), {
+            target: { value: mockCity.city }
+        });
+
+        expect(cityPlaceholder).toHaveValue(mockCity.city);
```

Comment:
These tests are good, and agreed that they also cover the logic inside ContextWrapper,  I would also prefer having separate tests for context (since you anyway split the logic to a different file). As this component grows you might have more logic in this component, which might not directly use utilize context.

You can use test the initial values of context and if state is updated correctly (similar to what you have done here)

Reply:
Verified the Context API with a child component. Now working on testing it with state..

Reply:
Updated testcase for state.